### DISCUSSION TITLE
"python.analysis.extraPaths" for write own pipeline

### DISCUSSION THREAD

**Discussion by MatchaChoco010** [bnpr/Malt#153] [Feedback & Feature Requests]
Hello!

I'm getting an import error when I try to edit the pipeline.
![image](https://user-images.githubusercontent.com/20170107/139588049-efef54b7-e148-4de5-9b43-5a968244124c.png)

If I add the following to "python.analysis.extraPaths" in vscode settings.json, the error disappears.
![image](https://user-images.githubusercontent.com/20170107/139588054-6a120d0b-6c15-4d94-918c-5cc12238f79c.png)
![image](https://user-images.githubusercontent.com/20170107/139588056-6b35ee9b-c07d-4726-8883-e528b4d30601.png)

```json

{
    "files.associations": {
        "*.glsl": "cpp"
    },
    "C_Cpp.default.includePath": ["C:\\Users\\orito\\AppData\\Roaming\\Blender Foundation\\Blender\\2.93\\scripts\\addons\\BlenderMalt\\.MaltPath\\Malt\\Shaders",""],
    "C_Cpp.default.forcedInclude": ["C:\\Users\\orito\\AppData\\Roaming\\Blender Foundation\\Blender\\2.93\\scripts\\addons\\BlenderMalt\\.MaltPath\\Malt\\Shaders\\Intellisense\\intellisense.glsl"],
    "C_Cpp.autoAddFileAssociations": true,
    "C_Cpp.default.cppStandard": "c++03",
    "C_Cpp.default.compilerPath": "",
    "C_Cpp.default.browse.limitSymbolsToIncludedHeaders": true,
    "C_Cpp.errorSquiggles": "Disabled",

    "python.analysis.extraPaths": [
        "C:\\Users\\orito\\AppData\\Roaming\\Blender Foundation\\Blender\\2.93\\scripts\\addons\\BlenderMalt\\.MaltPath",
        "C:\\Users\\orito\\AppData\\Roaming\\Blender Foundation\\Blender\\2.93\\scripts\\addons\\BlenderMalt\\.MaltPath\\Malt\\.Dependencies-39"
    ]
}
```

I think it would be useful to add this to the generated settings.json, what do you think?

**Comment by pragma37** [comment]
Good one!
The last time I tried I couldn't get it to work (I was probably using the wrong setting).

Implemented in 0914c94a697959196dcce5d5a02284d8b69b2c78

**Reply by MatchaChoco010** [reply]
Nice!
Thanks for the very quick implementation!
